common/src/main/java/io/netty/util/internal/logging/Log4J2LoggerFactory.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.util.internal.logging;
+
+import org.apache.logging.log4j.LogManager;
+
+public final class Log4J2LoggerFactory extends InternalLoggerFactory {
```

Comment:
> Making them all final singleton.

I suggest that doing it in a separate PR since it's not related to the title.

In addition, should the constructor of these factory classes be private? If so, it will break some user codes since InternalLoggerFactory suggests using ` InternalLoggerFactory.setDefaultFactory(new Log4JLoggerFactory());`